### PR TITLE
Added completion date and made patch for Completion Date in older tasks

### PR DESCRIPTION
**Pull request by arjund-06**
Signed-off-by: Arjun Dave <davearjun25@gmail.com>

### Functionality:
Shows completion date when the user submits Status as "Completed" while logging a task and older tasks which are completed have completion date as "XX/XX/XXXX".

### Solution:
Changed the firebase functions,
Added lines to create a CompletionDate field with value "Not yet Completed" while creating a new task and in log work added lines to update the same field to the current date on logging task as Completed.

### Risk level:
- [ ] high 
- [X] medium
- [ ] low

### How to test:
1. First, deploy the functions file in the code.
2. Use the patch to mark older Completed tasks with the date "XX/XX/XXXX"
3. ng serve
4. Create a new task and then log work and mark task as completed and voila Completion Date appears in the Task Detail page.

#### When the task isn't completed,

image

#### After Completing the task,

image

### CODE REVIEW

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]
1. can u check that u have used the correct formatter in index.js.
2. completion date position in task-detail is not correct. It should be at the top right corner and decrease the font size.
3. make a new file in app folder with patch name and click button there

**Comment by arjund-06** [comment]
One tiny error, will reopen

**Review by shivamgoyal7 — APPROVED** [review]
ship it

**Comment by Shrishti1802** [comment]
Please resolve merge conflicts

File: functions/patchAPI.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const db = admin.firestore();
+
+exports.addCompletionDatePatch = functions.https.onRequest((request, response) => {
+    cors(request, response, () => {
+        console.log(request);
+        var completionDate = request.body.data.CompletionDate;
+        var id;
+        var result;
+        var p1 = db.collection("Tasks").get().then((querySnapshot) => {
```
Comment (viv9k) [inline functions/patchAPI.js:14]: convert these into let and const type, according to the requirement, Do also use eslint for tracking these issue.

Comment (viv9k) [reply functions/patchAPI.js:14]: Name querySnapshot is not good, please use something understandable.

File: functions/patchAPI.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            querySnapshot.forEach((doc) => {
+                id = doc.data().Id;
+                console.log(id);
+                db.collection("Tasks").doc(id).update({
```
Comment (viv9k) [inline functions/patchAPI.js:18]: Use eslint to track the bug here

File: src/app/body/task-details/task-details.component.html [hunk 3]
```diff
@@ -16,8 +16,8 @@ <h5 class="active nav-link navbar-text" style="font-weight:bolder">Task Details<
                             <li class=" active pr-3 py-1 justify-content-between">
                                 <span type="button" class="material-icons text-light" data-bs-toggle="tooltip" data-bs-placement="top" title="Edit" *ngIf="(authService.afauth.user | async) && task.Status!=='Completed'" (click)="editTask()">edit</span>
                             </li>
-                            <li class=" active pr-1 py-1 justify-content-between" data-bs-toggle="tooltip" data-bs-placement="top" title="Delete" *ngIf="authService.afauth.user | async">
-                                <span type="button" class="material-icons text-light mr-3" (click)="deleteTask()">
+                            <li class=" active pr-1 py-1 justify-content-between">
+                                <span type="button" class="material-icons text-light mr-3" data-bs-toggle="tooltip" data-bs-placement="top" title="Delete" *ngIf="authService.afauth.user | async" (click)="deleteTask()">
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.html:20]: What is the need of doing change here.

File: src/app/body/task-details/task-details.component.html [hunk 4]
```diff
@@ -32,7 +32,11 @@ <h5 class="active nav-link navbar-text" style="font-weight:bolder">Task Details<
                             <div style="font-weight: bold;" class="mr-3">Date:</div>
                             <div id="creationDateTaskDescription">{{task.CreationDate}}</div>
                         </div>
-                        <div class="col-md-5 col-1"></div>
+                        <div class="col-md-4 col-1"></div>
+                        <div class="alert form-inline col-md-4 col-5 mb-5" style="font-size: 20px;" role="alert" *ngIf="task.Status == 'Completed'">
+                            <div style="font-weight: bold;" class="mr-3" style="color:#10be65">Completion Date <span style="color:white;">:</span></div>
+                            <div id="creationDateTaskDescription">{{task.CompletionDate}}</div>
```
Comment (viv9k) [inline src/app/body/task-details/task-details.component.html:38]: Make the UI little better

**Review by viv9k — CHANGES_REQUESTED** [review]
Complete the required changes asap

File: src/app/body/task-details/task-details.component.html (hunk 3 above)
Comment (arjund-06) [reply src/app/body/task-details/task-details.component.html:20]: @bighnesh0404 told me to do this...

File: src/app/app.module.ts [hunk 5]
```diff
@@ -63,11 +64,14 @@ import { RangeCardComponent } from "./body/my-dashboard/performance-chart/range-
...
     StatusIconsComponent,
     FilterTaskComponent,
+    FilterTaskComponent,
     NavbarComponent,
     ChartComponent,
     MyTasksComponent,
     PerformanceChartComponent,
-    RangeCardComponent
+    RangeCardComponent,
+    FilterTaskComponent,
```
Comment (viv9k) [inline src/app/app.module.ts:73]: Multiple addition of filter task components seen, please fix

**Review by viv9k — CHANGES_REQUESTED** [review]
Please fix the required changes

**Review by bighnesh0404 — APPROVED** [review]
Ship it

**Review by viv9k — APPROVED** [review]
ship it

**Pull request merged**